### PR TITLE
Allow apps to use different configs

### PR DESCRIPTION
**Pull request by anxolin**
This PR is the first step to separate the configurations for the different apps.

Creates 2 webpack config: one for v2 (default) , one for gp v1.

Now there's a refactored `getWebpackConfig` that do all the common work (creates webpack config based in some apps and config)

This PR includes some nice features like:

Each webpack entry config
- Decides wich env vars it needs in it's app
- Decides wich config it needs
- Decides wich apps it needs: for example v2 define explorer and the trade UI

We can still run gp-v1 and build, but now it's config doesn't need to containate other apps. We can define envs and vars that are really required in the APIs or modules used.

Future PRs could iterate and improve how we load the config. Right now we still load the whole YAML file, it would be nice to separate the apps further and have minimal config.

### CODE REVIEW

File: getWebpackConfig.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+      minifyURLs: true,
+    },
+  })
+}
+
+function getWebpackConfig({ apps = [], config = {}, envVars = {}, defineVars = {}, baseUrl = '/' } = {}) {
+  const { name: appTitle, templatePath, logoPath } = config
+  const isProduction = process.env.NODE_ENV == 'production'
+
+  assert(apps.length > 0, "At least one app it's required")
```
Comment (alfetopito) [inline getWebpackConfig.js:45]:
```suggestion
  assert(apps.length > 0, "At least one app is required")
```

File: webpack.config.js [hunk 2]
```diff
@@ -47,208 +65,36 @@ function getSelectedApps() {
...
-    filename: isProduction ? '[name].[contenthash].js' : '[name].js',
-    publicPath: baseUrl,
+module.exports = getWebpackConfig({
+  apps,
+  config,
+  baseUrl,
+  envVars: {
+    BASE_URL: baseUrl,
+    // // MOCK: Use mock or real API implementation
+    // MOCK: 'false',
```
Comment (alfetopito) [inline webpack.config.js:76]: I use this

File: webpack.config.js [hunk 3]
```diff
@@ -47,208 +65,36 @@ function getSelectedApps() {
...
+    // MOCK_WALLET: process.env.MOCK || 'false',
+    // MOCK_TOKEN_LIST: process.env.MOCK || 'false',
+    // MOCK_ERC20: process.env.MOCK || 'false',
```
Comment (alfetopito) [inline webpack.config.js:79]: I use this

File: webpack.config.js [hunk 4]
```diff
@@ -47,208 +65,36 @@ function getSelectedApps() {
...
+    // MOCK_WETH: process.env.MOCK || 'false',
+    // MOCK_DEPOSIT: process.env.MOCK || 'false',
+    // MOCK_EXCHANGE: process.env.MOCK || 'false',
+    // MOCK_WEB3: process.env.MOCK || 'false',
+    // MOCK_OPERATOR: process.env.MOCK || 'false',
```
Comment (alfetopito) [inline webpack.config.js:84]: I use this

File: webpack.config.js [hunk 5]
```diff
@@ -47,208 +65,36 @@ function getSelectedApps() {
...
+    // // AUTOCONNECT: only applies for mock implementation
+    // AUTOCONNECT: 'true',
+    // PRICE_ESTIMATOR_URL: process.env.PRICE_ESTIMATOR_URL || (isProduction && 'production') || 'develop',
+    // APP: null,
+    // APP_ID: null,
+    // INFURA_ID: null,
```
Comment (alfetopito) [inline webpack.config.js:90]: I use this

**Review by alfetopito — COMMENTED** [review]
I don't fully get this yet.
Leaving a comment for now.
Maybe it'll make more sense once we have one config for the explorer as well?

**Comment by anxolin** [comment]
> I don't fully get this yet.

We can discuss this today

**Review by alfetopito — APPROVED** [review]

**Comment by anxolin** [comment]
Merging to consolidate and let the footer point to the new repo. Go without saying, but @W3stside u can still comment

**Pull request merged**